packages/components/src/internal/components/domainproperties/list/ListDesignerPanels.tsx

```diff
@@ -106,6 +109,31 @@ class ListDesignerPanelsImpl extends React.PureComponent<Props & InjectedBaseDom
             });
     }
 
+    setKeyTypeForModel(): void {
+        const { model } = this.state;
+
+        if (!model.keyType) {
+            const fields = model.domain.fields;
+            const pkField = fields.find(i => i.isPrimaryKey);
+
+            if (pkField) {
+                const keyName = pkField.get('name');
+
+                let keyType;
+                // TODO this autoIncrement check doesn't seem to work for importing from JSON as the data type will just be INTEGER_TYPE and not AUTOINT_TYPE
```

Comment:
@RosalineP Note sure what to do about this case. The exported field defs don't have any indicator of the auto increment field type. I believe that settings is stored on the list domain model itself. Maybe we just open an issue for this so it doesn't block the merge of these PRs.

Reply:
Yeah, that's a shame. I can open the issue for this